jmespath-core/src/main/java/io/burt/jmespath/util/StringEscapeHelper.java

```diff
@@ -0,0 +1,103 @@
+package io.burt.jmespath.util;
+
+import java.util.Arrays;
+
+public class StringEscapeHelper {
+  private static final char NO_REPLACEMENT = 0xffff;
+
+  private final char[] unescapeMap;
+  private final char[] escapeMap;
+  private final boolean unescapeUnicodeEscapes;
+
+  public StringEscapeHelper(char... replacementPairs) {
+    this(false, replacementPairs);
+  }
+
+  public StringEscapeHelper(boolean unescapeUnicodeEscapes, char... replacementPairs) {
+    if (replacementPairs.length % 2 == 1) {
+      throw new IllegalArgumentException("Replacements must be even pairs");
+    }
+    this.unescapeUnicodeEscapes = unescapeUnicodeEscapes;
+    this.unescapeMap = createUnescapeMap(replacementPairs);
+    this.escapeMap = createEscapeMap(replacementPairs);
+  }
+
+  private static char[] createUnescapeMap(char[] pairs) {
+    char max = 0;
+    int i;
+    for (i = 0; i < pairs.length; i += 2) {
+      if (pairs[i] > max) {
+        max = pairs[i];
+      }
+    }
+    char[] replacementsMap = new char[max + 1];
+    Arrays.fill(replacementsMap, NO_REPLACEMENT);
+    for (i = 0; i < pairs.length; i += 2) {
+      replacementsMap[pairs[i]] = pairs[i + 1];
+    }
+    return replacementsMap;
+  }
+
+  private static char[] createEscapeMap(char[] pairs) {
```

Comment:
By adding `, int pairKeyIndex, int pairValueIndex` here, and using `pairs[i + pairKeyIndex]` and so on below, you wouldn't need to have the two almost identical methods here, but rather just call it like:
```java
unescapeMap = createEscapeMap(replacementPairs, 0, 1);
escapeMap = createEscapeMap(replacementPairs, 1, 0);
```
I guess technically you could specify only `pairKeyIndex` as an argument, and define `pairValueIndex = 1 - pairKeyIndex`, but that certainly seems overly clever :)